test/integration_tests/test_similarity.py

```diff
@@ -288,3 +296,15 @@ def test_should_not_do_faiss_index_scan_with_predicate(self):
 
         # Cleanup
         CatalogManager().drop_index_catalog_entry("testFaissIndexScanRewrite")
+
+    def test_end_to_end_index_scan_should_work_correctly_on_image_dataset(self):
+        create_index_query = """CREATE INDEX testFaissIndexImageDataset
+                                    ON testSimilarityImageDataset (DummyFeatureExtractor(data))
+                                    USING HNSW;"""
+        execute_query_fetch_all(create_index_query)
+        select_query = """SELECT data FROM testSimilarityImageDataset
+                            ORDER BY Similarity(DummyFeatureExtractor(Open("{}")), DummyFeatureExtractor(data))
+                            LIMIT 1;""".format(
+            self.img_path
+        )
+        execute_query_fetch_all(select_query)
```

Comment:
Can we make this test case more meaningful? First, it does not assert on anything. Second it seems there is only one image in the `testSimilarityImageDataset `,  it always returns that image I assume. For example, can we have 2+ images loaded and check it returns the similar one?